vmm/src/memory_manager.rs

```diff
@@ -151,6 +151,7 @@ pub struct MemoryManager {
     snapshot_memory_ranges: MemoryRangeTable,
     memory_zones: MemoryZones,
     log_dirty: bool, // Enable dirty logging for created RAM regions
+    arch_mem_regions: Vec<(GuestAddress, usize, RegionType)>,
```

Comment:
nit: Maybe at this point it should be a struct with named fields rather than a tuple